server/src/main/java/org/opensearch/cluster/routing/allocation/ExistingShardsAllocator.java

```diff
@@ -108,14 +112,26 @@ void allocateUnassigned(
      *
      * Allocation service will currently run the default implementation of it implemented by {@link ShardsBatchGatewayAllocator}
      */
-    default void allocateAllUnassignedShards(RoutingAllocation allocation, boolean primary) {
+    default BatchRunnableExecutor allocateAllUnassignedShards(RoutingAllocation allocation, boolean primary) {
```

Comment:
Do you need to return the `BatchRunnableExecutor` type here? Seems like you could avoid the weirdness of throwing an UnsupportedOperationException and a -1 timeout if this just returned a Runnable.

Reply:
Replied here to keep it simple - https://github.com/opensearch-project/OpenSearch/pull/14848#discussion_r1687456207

Reply:
I don't understand, isn't returning a `Runnable` here simpler?

Reply:
Yes, it would. Updated the code, will push